gtdynamics/universal_robot/Link.cpp

```diff
@@ -0,0 +1,75 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  Joint.cpp
+ * @brief Absract representation of a robot joint.
+ */
+
+#include "gtdynamics/universal_robot/Link.h"
+
+#include <gtsam/slam/expressions.h>
+
+#include <iostream>
+
+#include "gtdynamics/dynamics/Dynamics.h"
+#include "gtdynamics/statics/Statics.h"
+
+namespace gtdynamics {
+
+gtsam::Vector6_ Link::wrenchConstraint(
+    const std::vector<DynamicsSymbol>& wrench_keys, uint64_t t,
+    const boost::optional<gtsam::Vector3>& gravity) const {
+  // Collect wrenches to implement L&P Equation 8.48 (F = ma)
+  std::vector<gtsam::Vector6_> wrenches;
+
+  // Coriolis forces.
+  gtsam::Vector6_ twist(internal::TwistKey(id(), t));
+  gtsam::Vector6_ wrench_coriolis(
+      std::bind(Coriolis, inertiaMatrix(), std::placeholders::_1,
```

Comment:
elegance detector firing

Reply:
I don't get it. Is the function too long?

Reply:
Same as with the other expressions. If Coriolis was a functor, you wouldn’t have to do this placeholder dance

Reply:
Added a TODO here